src/internal_modules/roc_core/heap_arena.h

```diff
@@ -45,15 +60,22 @@ class HeapArena : public IArena, public NonCopyable<> {
     //! Deallocate previously allocated memory.
     virtual void deallocate(void*);
 
+    //! Get number of guard failures.
+    size_t num_guard_failures() const;
```

Comment:
nitpick: let's move this method right after num_allocations(), another counter getter used in tests.